target/arc/op_helper.c

```diff
@@ -428,8 +428,13 @@ overflow_add_flag(target_ulong dest, target_ulong b, target_ulong c,
                   uint8_t size)
 {
     dest >>= (size - 1);
-    b >>= (size - 1);
-    c >>= (size - 1);
+    b    >>= (size - 1);
+    c    >>= (size - 1);
+
```

Comment:
Please add a comment like the one you mentioned in the issue.

```
This will only work if after the shift, there is nothing to the left of the 
bit originally at the 'size'th place. This does not hold true when we are
dealing with 32 bit overflow calculations in a 64 bit architecture, if one
of the numbers is negative (the most significant 32 bits will all be 1).
```